src/OrchardCore/OrchardCore.Navigation.Core/NavigationHelper.cs

```diff
@@ -176,7 +191,7 @@ private static dynamic GetHighestPrioritySelectedMenuItem(dynamic parentShape)
                     }
                     else // found more selected: tie break required.
                     {
-                        if (item.Priority > result.Priority)
+                        if (item.SelectionPriority > result.SelectionPriority)
```

Comment:
menuItemShape is not set up with Priority, it should use SelectionPriority Or Item.Priority